tests/cpp/test_topk_device_func.cpp

```diff
@@ -0,0 +1,324 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+
+#include <gtest/gtest.h>
+
+#include <tests/cpp/topk_test_helper.h>
+#include <tests/cpp/utils.h>
+#include <tests/cpp/validator.h>
+
+#include <ATen/cuda/CUDAContext.h>
+#include <algorithm>
+#include <cstdint>
+#include <random>
+#include <vector>
+
+namespace nvfuser {
+
+using TopkDeviceFuncTest = NVFuserTest;
+
+template <typename DataT>
+std::vector<DataT> getVector(at::Tensor tensor) {
+  NVF_ERROR_EQ(tensor.dim(), 1);
+  if (tensor.dtype() == at::kBFloat16) {
+    tensor = tensor.to(at::kFloat);
+  }
+  auto cpu_tensor = tensor.cpu();
+  auto total_elements = tensor.size(0);
+  return std::vector<DataT>(
+      cpu_tensor.data_ptr<DataT>(),
+      cpu_tensor.data_ptr<DataT>() + total_elements);
+}
+
+// Helper function to validate topk correctness
+template <typename DataT>
+bool validateTopkOrder(
+    const at::Tensor& input_tensor,
+    const at::Tensor& values_tensor,
+    const at::Tensor& indices_tensor,
+    int k,
+    bool largest = true) {
+  NVF_ERROR_EQ(cudaDeviceSynchronize(), cudaSuccess);
+
+  auto input_data = getVector<DataT>(input_tensor);
+  auto output_values = getVector<DataT>(values_tensor);
+  auto output_indices = getVector<int64_t>(indices_tensor);
+
+  // Check that we have k valid results
+  if (output_values.size() < k || output_indices.size() < k) {
+    return false;
+  }
+
+  // Check valid indices range
+  for (int i = 0; i < k; i++) {
+    if (output_indices[i] < 0 || output_indices[i] >= input_data.size()) {
+      return false;
+    }
+  }
+
+  // Check values match indices
+  for (int i = 0; i < k; i++) {
+    if (output_values[i] != input_data[output_indices[i]]) {
+      return false;
+    }
+  }
+
+  // Check sorting order of the k elements
+  for (int i = 1; i < k; i++) {
```

Comment:
```suggestion
  for (size_t i = 1; i < k; i++) {
```
**[Suggested by AI]** The code change involves modifying the type of the loop counter variable `i` from `int` to `size_t`.

**Summary:**

This change likely addresses a potential issue where the loop counter `i` could exceed the maximum value representable by an `int`, especially if `k` is large.  `size_t` is an unsigned integer type guaranteed to be large enough to represent the size of any object in memory, making it a more appropriate choice for indexing into arrays or containers, and preventing potential integer overflow issues. This fix resolves a build error that was likely caused by this overflow or a type mismatch when comparing `i` and `k`.